### PR TITLE
A few little fixes

### PR DESCRIPTION
**Pull request by sousoux**
1) There is no way to disable colors at present. This patch helps one do that.

2) Possibility of hang since callback is not called if socket is destroyed

3) Currenly no was to require('bundle:/absolute/path/to/something')

This one requires a look. Should it be bundle://absolute?

### CODE REVIEW

**Comment by zhaozg** [comment]
Like point 2

**Comment by creationix** [comment]
> 1) There is no way to disable colors at present. This patch helps one do that.

Sure there is, just set it to `false`.  Try running `luvit -n` on the command line to see it in action.

**Comment by sousoux** [comment]
True. That is dumb. The other two are valid though I think.

File: deps/net.lua [hunk 1]
```diff
@@ -137,7 +137,7 @@ end
 
 function Socket:shutdown(callback)
   if self.destroyed == true then
-    return
+    return callback()
```
Comment (creationix) [inline deps/net.lua]: Good catch.

File: deps/require.lua [hunk 2]
```diff
@@ -229,9 +229,10 @@ end
 
 function Module:resolve(name)
   assert(name, "Missing name to resolve")
-  if name:byte(1) == 46 then -- Starts with "."
+  local debundled_name = name:match("^bundle:(.*)") or name
+  if debundled_name:byte(1) == 46 then -- Starts with "."
     return fixedRequire(pathJoin(self.dir, name))
-  elseif name:byte(1) == 47 then -- Starts with "/"
+  elseif debundled_name:byte(1) == 47 then -- Starts with "/"
```
Comment (creationix) [inline deps/require.lua]: Nice fix.  I was wondering why this didn't work.

**Comment by creationix** [comment]
> That is dumb.

Well, I needed a value that means to use whatever color scheme is default for the platform.  I have logic elsewhere to try and detect if the console supports 256 colors or just 16 colors.

There was a bug before where it was impossible to disable colors, but it was fixed by checking for exactly `nil` instead of just any falsy value.

**Comment by sousoux** [comment]
I meant I was being dumb ... not you!
Thanks for merging the other two

**Pull request merged**